peer.go

```diff
@@ -213,8 +208,8 @@ func (p *peer) Start() error {
 	// Before we launch any of the helper goroutines off the peer struct,
 	// we'll first ensure proper adherence to the p2p protocol. The init
 	// message MUST be sent before any other message.
-	readErr := make(chan error, 1)
-	msgChan := make(chan lnwire.Message, 1)
```

Comment:
What if read message return an error? In this case it probably might lead to goroutine waiting to write in `msgChan`, but nobody will read from it, because we exited from start.